core-api/src/main/java/com/optimizely/ab/internal/DefaultLRUCache.java

```diff
@@ -93,53 +84,32 @@ public T lookup(String key) {
         }
 
         synchronized (lock) {
-            if (hashMap.containsKey(key)) {
-                ItemWrapper item = hashMap.get(key);
+            if (linkedHashMap.containsKey(key)) {
+                ItemWrapper item = linkedHashMap.get(key);
                 Long nowMs = new Date().getTime();
 
                 // ttl = 0 means items never expire.
                 if (timeoutMillis == 0 || (nowMs - item.timestamp < timeoutMillis)) {
-                    linkedList.remove(item);
-                    linkedList.addFirst(item);
                     return item.value;
                 }
 
-                // If item is stale, remove it.
-                linkedList.remove(item);
-                hashMap.remove(item.key);
+                linkedHashMap.remove(key);
             }
             return null;
         }
     }
 
-    public T peek(String key) {
```

Comment:
It's surprising that they do not have this method in LinkedHashMap.  It should be ok without this since it's mostly for testing/debugging.